aiodiscover/discovery.py

```diff
@@ -139,11 +139,19 @@ async def _async_get_nameservers(
         """Get nameservers to query."""
         all_nameservers = list(sys_network_data.nameservers)
         router_ip = sys_network_data.router_ip
-        assert router_ip is not None
-        if router_ip not in all_nameservers:
-            neighbours = await sys_network_data.async_get_neighbors([router_ip])
-            if router_ip in neighbours:
-                all_nameservers.insert(0, router_ip)
+        # if none of the nameservers are in the same network as the router
+        # insert it to the list of nameservers
+        add_router = True
+        for nameserver in all_nameservers:
```

Comment:
The loop variable `nameserver` isn't used in the loop. This loop doesn't make sense